contracts/handlers/AcrossMulticallHandler.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "../SpokePool.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+/**
+ * @title Across Multicall contract that allows a user to specify a series of calls that should be made by the handler
+ * via the message field in the deposit.
+ * @dev This contract makes the calls blindly. The caller should ensure that the tokens recieved by the handler are completely consumed.
+ */
+contract AcrossMulticall is AcrossMessageHandler, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+    using Address for address payable;
+
+    struct Call {
+        address target;
+        bytes callData;
+        uint256 value;
+    }
+
+    error CallReverted(uint256 index);
+    error NotSelf();
+
+    /**
+     * @notice Main entrypoint for the handler called by the SpokePool contract.
+     * @dev This will execute all calls encoded in the msg. The caller is responsible for making sure all tokens are
+     * drained from this contract by the end of the series of calls. If not, they can be stolen.
+     * A drainRemainingTokens call can be included as a way to drain any remaining tokens from this contract.
+     * @param message abi encoded array of Call structs, containing a target, callData, and value for each call that
+     * the contract should make.
+     */
+    function handleV3AcrossMessage(
+        address,
+        uint256,
+        address,
+        bytes memory message
+    ) external nonReentrant {
+        Call[] memory calls = abi.decode(message, (Call[]));
```

Comment:
Is there any validation we want to do here on converting `message` into `Call[]`

Reply:
What sort of validation would you suggest?

My understanding is that abi.decode will cause a revert if the format is wrong.